lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -195,18 +213,73 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
   defp match_struct(hint, ctx) do
     for {:module, module, name, documentation} <- match_alias(hint, ctx, true),
         has_struct?(module),
+        not is_exception?(module),
         do: {:module, module, name, documentation}
   end
 
   defp has_struct?(mod) do
-    Code.ensure_loaded?(mod) and function_exported?(mod, :__struct__, 1) and
-      not function_exported?(mod, :exception, 1)
+    Code.ensure_loaded?(mod) and function_exported?(mod, :__struct__, 1)
+  end
+
+  defp is_exception?(mod) do
+    Code.ensure_loaded?(mod) and function_exported?(mod, :exception, 1)
   end
 
   defp match_module_member(mod, hint, ctx) do
     match_module_function(mod, hint, ctx) ++ match_module_type(mod, hint, ctx)
   end
 
+  defp match_in_struct_fields_or_local_or_var(hint, ctx) do
+    case expand_struct_fields(ctx) do
+      {:ok, struct, fields} ->
+        for {field, default} <- fields,
+            name = Atom.to_string(field),
+            ctx.matcher.(name, hint),
+            do: {:in_struct_field, struct, field, default}
+
+      _ ->
+        match_local_or_var(hint, ctx)
+    end
+  end
+
+  defp expand_struct_fields(ctx) do
+    with {:ok, quoted} <- Code.Fragment.container_cursor_to_quoted(ctx.fragment),
+         {aliases, pairs} <- find_struct_fields(quoted) do
+      mod_name = Enum.join(aliases, ".")
+      mod = expand_alias(mod_name, ctx)
+
+      fields =
+        if has_struct?(mod) do
+          # Remove the keys that have already been filled, and internal keys
+          Map.from_struct(mod.__struct__)
+          |> Map.drop(Keyword.keys(pairs))
+          |> Map.reject(fn {key, _} ->
+            key
+            |> Atom.to_string()
+            |> String.match?(~r/__.*__/)
```

Comment:
We should remove any key with underscore. That's the convention for anything private in Elixir. `__...__` is about reflection.

Reply:
```suggestion
            |> String.starts_with?("_")
```